File: 0229-majority-element-ii/0229-majority-element-ii.py
from collections import Counter

class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        n=len(nums)//3
        b=Counter(nums)

        L=[]
        for i in b.keys():
            if b.get(i)>n:
                L.append(i)
        return L

File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        Map<Integer, Integer> dict = new HashMap<>();
        for (int i : nums) {
            dict.put(i, dict.getOrDefault(i, 0) + 1);
        }
        
        List<Integer> res = new ArrayList<>();
        int compare = nums.length / 3;
        for (Map.Entry<Integer, Integer> entry : dict.entrySet()) {
            if (entry.getValue() > compare) {
                res.add(entry.getKey());
            }
        }
        
        return res;
    }
}